react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,98 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null)
+    }],
+    stepNumber: 0,
+    player: 'X'
+  };
+
+  changeTurn = () => this.state.player === 'X' ? 'O' : 'X';
+
+  jumpTo(step) {
+    if (this.state.player === 'X' && step % 2) {
+      this.setState({ stepNumber: step, player: 'X' });
+    }
+    if (this.state.player === 'O' && !(step % 2)) {
+      this.setState({ stepNumber: step, player: 'O' });
+    }
+  }
+
+  handleClick = i => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[this.state.stepNumber];
+    const squares = [...current.squares];
+
+    if (this.calculateWinner(squares) || squares[i]) {
```

Comment:
Use semantic conditions for this. Good code it's not only about few lines, readability is a must

```
const isThereAWinner = this.calculateWinner(squares);
const isSquareFilled = squares[i];
```